activerecord/test/cases/transactions_test.rb

```diff
@@ -520,7 +520,7 @@ def test_force_savepoint_in_nested_transaction
 
       begin
         Topic.transaction requires_new: true do
-          @first.happy = false
+          @first.approved = false
```

Comment:
The assertion below is `assert_predicate @first.reload, :approved?` so we need to set the `approved` column and not the `happy` column.

Reply:
It turns out there's no `happy` column, but the NoMethodError for `happy=` was being rescued below. 🙃

Reply:
Ahh the perils of using a blanket `rescue`. Should I update the test to raise a specific error? That would at least prevent future errors like this or typos.

Reply:
The tests for real transactions often use `ActiveRecord::Rollback` for this purpose, e.g.:

https://github.com/rails/rails/blob/98135434d29bcafbc36cfcb53e1370c5c044cc8a/activerecord/test/cases/transactions_test.rb#L825-L828

That doesn't currently work for savepoints, but it will after https://github.com/rails/rails/pull/39947, so you could change these tests to use `raise ActiveRecord::Rollback` instead of `raise` there too.